sherpa/astro/io/pyfits_backend.py

```diff
@@ -163,62 +206,67 @@ def _try_vec(hdu, name, size=2, dtype=SherpaFloat, fix_type=False):
         col = col.astype(dtype)
 
     if col is None:
-        return numpy.array([None]*size)
+        return numpy.array([None] * size)
 
     return col
 
+
 def _require_col(hdu, name, dtype=SherpaFloat, fix_type=False):
     col = _try_col(hdu, name, dtype, fix_type)
     if col is None:
         raise IOErr('reqcol', name, hdu._file.name)
     return col
 
+
 def _require_tbl_col(hdu, name, dtype=SherpaFloat, fix_type=False):
     col = _try_tbl_col(hdu, name, dtype, fix_type)
     if len(col) > 0 and col[0] is None:
         raise IOErr('reqcol', name, hdu._file.name)
     return col
 
+
 def _require_vec(hdu, name, size=2, dtype=SherpaFloat, fix_type=False):
     col = _try_vec(hdu, name, size, dtype, fix_type)
     if numpy.equal(col, None).any():
         raise IOErr('reqcol', name, hdu._file.name)
     return col
 
+
 def _try_col_or_key(hdu, name, dtype=SherpaFloat, fix_type=False):
     col = _try_col(hdu, name, dtype, fix_type)
     if col is not None:
         return col
     return _try_key(hdu, name, fix_type, dtype)
 
+
 def _try_vec_or_key(hdu, name, size, dtype=SherpaFloat, fix_type=False):
     col = _try_col(hdu, name, dtype, fix_type)
     if col is not None:
         return col
-    return numpy.array([_try_key(hdu, name, fix_type, dtype)]*size)
+    return numpy.array([_try_key(hdu, name, fix_type, dtype)] * size)
 
 
-
-## Read Functions ##
+# Read Functions
 
 def read_table_blocks(arg, make_copy=False):
 
     filename = ''
     hdus = None
-    if type(arg) is pyfits.HDUList:
+    if type(arg) is fits.HDUList:
         filename = arg[0]._file.name
         hdus = arg
     elif type(arg) in (str, unicode, numpy.str_) and is_binary_file(arg):
```

Comment:
There are more robust ways of checking if a variable is a string. It is probably not worth trying to employ patterns that are both Python 2 and Python 3- compliant at this stage, as this would most likely require the addition of more OTS packages (e.g. `six` or `future`) we still cannot add.